src/CodeGenerator/src/Generator/CodeGenerator/TypeGenerator.php

```diff
@@ -117,58 +117,25 @@ public function getPhpType(Shape $shape): array
 
         if ($shape instanceof ListShape) {
             $listMemberShape = $shape->getMember()->getShape();
-            if ($listMemberShape instanceof StructureShape) {
-                $memberClassNames[] = $className = $this->namespaceRegistry->getObject($listMemberShape);
-
-                return ['array', $className->getName() . '[]', $memberClassNames];
-            }
-            if ($listMemberShape instanceof ListShape) {
-                $listMemberShapeLevel2 = $listMemberShape->getMember()->getShape();
-                if ($listMemberShapeLevel2 instanceof StructureShape) {
-                    $memberClassNames[] = $className = $this->namespaceRegistry->getObject($listMemberShapeLevel2);
-
-                    return ['array', $className->getName() . '[][]', $memberClassNames];
-                }
-
-                if (!empty($listMemberShapeLevel2->getEnum())) {
-                    $memberClassNames[] = $memberClassName = $this->namespaceRegistry->getEnum($listMemberShapeLevel2);
-                    $doc = 'list<list<' . $memberClassName->getName() . '::*>>';
-                } else {
-                    $doc = $this->getNativePhpType($listMemberShapeLevel2->getType()) . '[][]';
-                }
-
-                return ['array', $doc, $memberClassNames];
-            }
-
-            if (!empty($listMemberShape->getEnum())) {
-                $memberClassNames[] = $memberClassName = $this->namespaceRegistry->getEnum($listMemberShape);
-                $doc = 'list<' . $memberClassName->getName() . '::*>';
+            [$type, $doc, $memberClassNames] = $this->getPhpType($listMemberShape);
+            if (\substr($doc, -3) === '::*') {
+                $doc = "list<$doc>";
             } else {
-                $doc = $this->getNativePhpType($listMemberShape->getType()) . '[]';
+                $doc .= '[]';
```

Comment:
I whish using `list<$doc>` everywhere, but incompatible with `array` documented in `@param` input constructor.
This would requires a bigger refactor